### PR TITLE
Dark mode commeth

### PR DESCRIPTION
**Pull request by asutula**
Dark mode support, switch you preference from a button in the header.

### CODE REVIEW

File: packages/web/app/[team]/(team)/layout.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ export default async function LayoutTeam({
 }) {
   return (
     <div className="flex flex-1 flex-col">
-      <header className="sticky top-0 flex flex-col space-y-4 border-b bg-white px-4 py-3">
```
Comment (asutula) [inline packages/web/app/[team]/(team)/layout.tsx:10]: Anywhere you set a background color, to be dark mode compatible, you must use colors that are specified/overridden by the shadui settings in `tailwind.config.js` (I think, still wrapping my head around how this all works).

File: packages/web/app/_components/footer.tsx [hunk 2]
```diff
@@ -4,7 +4,7 @@ import logo from "@/public/white.svg";
 
 export default function Footer() {
   return (
-    <footer className="text-neonblue bg-black py-10 font-light text-white">
+    <footer className="text-neonblue bg-foreground py-10 font-light text-white dark:border-t dark:bg-background">
```
Comment (asutula) [inline packages/web/app/_components/footer.tsx:7]: In the footer, it is black when in light mode, but we don't want it to be white in dark mode, so I set it to stay black here and add a top border.

File: packages/web/app/_components/theme-switcher.tsx
Comment (asutula) [inline packages/web/app/_components/theme-switcher.tsx:1]: The theme switcher button.

File: packages/web/app/layout.tsx [hunk 3]
```diff
@@ -76,23 +78,31 @@ export default async function RootLayout({
...
+                      <NavPrimary />
+                      <Profile />
+                      <ThemeSwitcher />
+                    </div>
+                  </header>
+                  <div className="flex flex-1 flex-col">{children}</div>
+                  <Footer />
+                  <Toaster />
+                </TRPCReactProvider>
+              </ThemeProvider>
```
Comment (asutula) [inline packages/web/app/layout.tsx:105]: We need to wrap our app in the `ThemeProvider`.

File: packages/web/components/mesa-svg.tsx [hunk 4]
```diff
@@ -10,7 +26,7 @@ export default function MesaSvg() {
         fillRule="evenodd"
         clipRule="evenodd"
         d="M5.12366 4.56488C5.20594 4.16302 5.56357 3.85712 5.98533 3.85712H9.47261C9.93884 3.85712 10.1465 3.64743 10.4236 3.24051L11.9604 0.430052C12.1052 0.165416 12.3884 0 12.6969 0H25.5301C25.8606 0 26.1355 0.247737 26.162 0.567468C26.6782 6.79843 27.3326 11.022 31.4761 14.845C31.9108 15.2461 31.6328 16 31.0329 16H0.655819C0.0550865 16 -0.223428 15.2441 0.211887 14.8424C3.17796 12.1053 4.36631 9.11091 4.99402 5.23535C5.02545 5.04128 5.06395 4.85465 5.10259 4.66731C5.10962 4.63321 5.11666 4.59908 5.12366 4.56488Z"
-        fill="black"
+        fill={theme === "dark" ? "#fff" : "#000"}
```
Comment (asutula) [inline packages/web/components/mesa-svg.tsx:29]: Color the mesa svg depending on the theme.

File: packages/web/components/ui/dropdown-menu.tsx
Comment (asutula) [inline packages/web/components/ui/dropdown-menu.tsx:1]: This is really unrelated. I was having a problem getting things working and tried updating this shadui component to see if it fixed things. It wasn't the cause of my problem, but left this update in place anyway.

**Comment by joewagner** [comment]
Very excited about this!
My only thought is that our dark mode contrasts don't quite look right.  If merging this and addressing that later is best I'm ok with that.
Specifically the borders of all the boxes etc... are not light enough.  The WCAG docs explain my thinking https://www.w3.org/WAI/WCAG22/quickref/?versions=2.0#non-text-content

**Comment by joewagner** [comment]
Looks like we might need to run prettier:fix

**Review by joewagner — APPROVED** [review]
😎

**Comment by asutula** [comment]
Closing this for now, we can reopen if/when we decide to do dark/light toggle.
